viewing_party/party.py

```diff
@@ -1,23 +1,223 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    # wave part 1
+    if title is None or genre is None or rating is None:
+        return None
+    
+    else:
+        dict = {}
+        dict["title"] = title
+        dict["genre"] = genre
+        dict["rating"] = rating
+        return dict
+
+# wave 1 part 2
+
+def add_to_watched(user_data, movie):
+    list_of_dicts = []
+    user_data["watched"].append(movie)
+    return user_data
+
+#wave 1 part 3
+def add_to_watchlist(user_data, movie):
+    list_of_dicts = []
+    user_data["watchlist"].append(movie)
+    return user_data
+
+#wave  1 part 4
+
+def watch_movie(user_data, title):
+
+    #there is a dict user_data with key watchlist and watched 
+    #watchlist is a key with value list of dicts
+    # watched is a key with value list
+    
+    temp_list_of_dict = user_data["watchlist"] 
+    for i in range(0, len(temp_list_of_dict)):
+        temp_dict = temp_list_of_dict[i] 
+        temp_string = temp_dict["title"] 
+        temp_list_watched=user_data["watched"] 
+        if temp_string is title:
+            user_data["watchlist"].remove(temp_dict)
+            user_data["watched"].append(temp_dict)
+    return user_data

```

Comment:
- Consider using a for-each loop instead of a for-range loop here. The index is not important.
- When comparing strings use `==` instead of `is`. We pretty much only ever use `is` when checking whether something is `None`.
- This works in this particular case because we only expect the user to have one copy of each movie in their watchlist. But in general, try to avoid modifying a data structure as we're doing a for loop over it; very strange and hard to debug behavior can result. Consider instead using the loop to identify what needs to be removed, then remove it after the loop.